### PR TITLE
Adds the remote client address to server spans for gRPC

### PR DESCRIPTION
**Pull request by ewhauser**
This address the feature request for #185

### CODE REVIEW

File: brave-grpc/src/main/java/com/github/kristofa/brave/grpc/BraveGrpcServerInterceptor.java [hunk 1]
```diff
@@ -90,7 +93,14 @@ public String getSpanName() {
 
         @Override
         public Collection<KeyValueAnnotation> requestAnnotations() {
-            return Collections.emptyList();
+            SocketAddress socketAddress = call.attributes().get(ServerCall.REMOTE_ADDR_KEY);
+            if (socketAddress != null) {
+                KeyValueAnnotation remoteAddrAnnotation = KeyValueAnnotation.create(
```
Comment (adriancole) [inline brave-grpc/src/main/java/com/github/kristofa/brave/grpc/BraveGrpcServerInterceptor.java]: in zipkin, it would be more normal to add a binary annotation of type "ca" (client address). This is done via `ServerTracer.setServerReceived(int ipv4, int port, @Nullable String clientService)`, though maybe it isn't easy to get at here..

Comment (ewhauser) [inline brave-grpc/src/main/java/com/github/kristofa/brave/grpc/BraveGrpcServerInterceptor.java]: The [ServerRequestInterceptor](https://github.com/openzipkin/brave/blob/master/brave-core/src/main/java/com/github/kristofa/brave/ServerRequestInterceptor.java#L49) makes the call to `setServerReceived()`.  Because of the way the [ServerRequestAdapter](https://github.com/openzipkin/brave/blob/master/brave-core/src/main/java/com/github/kristofa/brave/ServerRequestAdapter.java) interface is setup, there is not a hook to set the client address annotation prior to `setServerReceived()` being called.

If you think it makes sense, I can add an additional method to `ServerRequestAdapter` that allows implementors to optionally supply the client info so it can be used by `ServerRequestInterceptor`.  We could also use this to enhance the `HttpServerRequestAdapter` as well using the various methods to pull client IPs ([ServletRequest.html#getRemoteAddr()](i.e. http://docs.oracle.com/javaee/6/api/javax/servlet/ServletRequest.html#getRemoteAddr())and well known proxy headers such as `X-Forwarded-For`)

Comment (ewhauser) [inline brave-grpc/src/main/java/com/github/kristofa/brave/grpc/BraveGrpcServerInterceptor.java]: I'd probably do that under a separate PR since it's a bigger change that affects more than just this issue

Comment (adriancole) [inline brave-grpc/src/main/java/com/github/kristofa/brave/grpc/BraveGrpcServerInterceptor.java]: sounds good.

File: brave-grpc/src/test/java/com/github/kristofa/brave/grpc/BraveGrpcInterceptorsTest.java [hunk 2]
```diff
@@ -150,16 +150,22 @@ public void noSamplesWhenSamplingDisabled() throws Exception {
...
         assertThat(spans.size(), is(equalTo(2)));
-        assertThat(spans.get(0).getTrace_id(), is(equalTo(spans.get(1).getTrace_id())));
-        validateSpan(spans.get(0), Arrays.asList("ss", "sr"));
+
+        Span serverSpan = spans.get(0);
+        assertThat(serverSpan.getTrace_id(), is(equalTo(spans.get(1).getTrace_id())));
+        validateSpan(serverSpan, Arrays.asList("ss", "sr"));
+
+        //Server spans should have the GRPC_REMOTE_ADDR binary annotation
+        Optional<BinaryAnnotation> binaryAnnotation =
```
Comment (adriancole) [inline brave-grpc/src/test/java/com/github/kristofa/brave/grpc/BraveGrpcInterceptorsTest.java]: the value of this would be consistent, right? It might be nice to show what it is (ex ip:port)

here's some assertj slickness..

```java
    assertThat(span.binaryAnnotations)
        .filteredOn(b -> b.key.equals(Constants.SERVER_ADDR))
        .extracting(b -> new String(b.value))
        .containsOnly("host:port");
```

Comment (adriancole) [inline brave-grpc/src/test/java/com/github/kristofa/brave/grpc/BraveGrpcInterceptorsTest.java]: obv I used the wrong key, here.

**Comment by adriancole** [comment]
one test tidy if you don't mind, then let's merge this! thx

also highly appreciate if you can take a stab at formalizing the "CA" thing as you mentioned (in a different PR)

**Comment by ewhauser** [comment]
Done and done.  Will track "CA" under #205

**Comment by adriancole** [comment]
nicely done!
